ImportError: cannot import name 'file_hash' from 'pooch.utils'

**Description of the problem**

As Pooch was updated today, I get an import error for 'file_hash'. See also https://github.com/cgre-aachen/gemgis/runs/3401104133?check_suite_focus=true#step:6:37. Was it maybe moved?
<!--
Please be as detailed as you can when describing an issue. The more information
we have, the easier it will be for us to track this down.
-->


**Full error message**

```
ImportError: cannot import name 'file_hash' from 'pooch.utils'
```



**System information**

* Operating system: Win 10
* Python installation (Anaconda, system, ETS): Anaconda
* Version of Python: 3.9
* Version of this package: latest from pip
* If using conda, paste the output of `conda list` below:

<details>
<summary>output of conda list</summary>
<pre>

PASTE OUTPUT OF CONDA LIST HERE

</pre>
</details>

The import path change has broken the import of scikit-image as well 😨 

Specifically, I see the error coming from `skimage/data/__init__.py:64: ImportError`


<details>

```

    """Standard test images.

    For more images, see

     - http://sipi.usc.edu/database/database.php

    """
    from distutils.version import LooseVersion
    from warnings import warn
    import numpy as np
    import shutil

    from ..util.dtype import img_as_bool
    from ._binary_blobs import binary_blobs
    from ._registry import registry, legacy_registry, registry_urls

    from .. import __version__

    import os.path as osp
    import os

    __all__ = ['data_dir',
               'download_all',
               'astronaut',
               'binary_blobs',
               'brain',
               'brick',
               'camera',
               'cat',
               'cell',
               'cells3d',
               'checkerboard',
               'chelsea',
               'clock',
               'coffee',
               'coins',
               'colorwheel',
               'eagle',
               'grass',
               'gravel',
               'horse',
               'hubble_deep_field',
               'human_mitosis',
               'immunohistochemistry',
               'kidney',
               'lbp_frontal_face_cascade_filename',
               'lily',
               'lfw_subset',
               'logo',
               'microaneurysms',
               'moon',
               'page',
               'text',
               'retina',
               'rocket',
               'shepp_logan_phantom',
               'skin',
               'stereo_motorcycle']

    legacy_data_dir = osp.abspath(osp.dirname(__file__))
    skimage_distribution_dir = osp.join(legacy_data_dir, '..')

    try:
>       from pooch.utils import file_hash
E       ImportError: cannot import name 'file_hash' from 'pooch.utils' (/opt/django-project/.tox/test-rgd-client/lib/python3.8/site-packages/pooch/utils.py)

LooseVersion = <class 'distutils.version.LooseVersion'>
__all__    = ['data_dir', 'download_all', 'astronaut', 'binary_blobs', 'brain', 'brick', ...]
__builtins__ = <builtins>
__cached__ = '/opt/django-project/.tox/test-rgd-client/lib/python3.8/site-packages/skimage/data/__pycache__/__init__.cpython-38.pyc'
__doc__    = 'Standard test images.\n\nFor more images, see\n\n - http://sipi.usc.edu/database/database.php\n\n'
__file__   = '/opt/django-project/.tox/test-rgd-client/lib/python3.8/site-packages/skimage/data/__init__.py'
__loader__ = <_frozen_importlib_external.SourceFileLoader object at 0x7f5cd669df10>
__name__   = 'skimage.data'
__package__ = 'skimage.data'
__path__   = ['/opt/django-project/.tox/test-rgd-client/lib/python3.8/site-packages/skimage/data']
__spec__   = ModuleSpec(name='skimage.data', loader=<_frozen_importlib_external.SourceFileLoader object at 0x7f5cd669df10>, origin=....py', submodule_search_locations=['/opt/django-project/.tox/test-rgd-client/lib/python3.8/site-packages/skimage/data'])
__version__ = '0.18.2'
binary_blobs = <function binary_blobs at 0x7f5cc64d68b0>
img_as_bool = <function img_as_bool at 0x7f5cc64b5a60>
legacy_data_dir = '/opt/django-project/.tox/test-rgd-client/lib/python3.8/site-packages/skimage/data'
legacy_registry = {'data/astronaut.png': '88431cd9653ccd539741b555fb0a46b61558b301d4110412b5bc28b5e3ea6cb5', 'data/brick.png': '7966caf3...3d5690fc7e3648f2795d730c23a', 'data/cell.png': '8d23a7fb81f7cc877cd09f330357fc7f595651306e84e17252f6e0a1b3f61515', ...}
np         = <module 'numpy' from '/opt/django-project/.tox/test-rgd-client/lib/python3.8/site-packages/numpy/__init__.py'>
os         = <module 'os' from '/usr/local/lib/python3.8/os.py'>
osp        = <module 'posixpath' from '/usr/local/lib/python3.8/posixpath.py'>
registry   = {'color/tests/ciede2000_test_data.txt': '2e005c6f76ddfb7bbcc8f68490f1f7b4b4a2a4b06b36a80c985677a2799c0e40', 'color/tes...24c7', 'color/tests/data/lab_array_d50_2.npy': '4aa03b7018ff7276643d3c082123cf07304f9d8d898ae92a5756a86955de4faf', ...}
registry_urls = {'data/brain.tiff': 'https://gitlab.com/scikit-image/data/-/raw/master/brain.tiff', 'data/cells3d.tif': 'https://gitla...eagle.png', 'data/kidney.tif': 'https://gitlab.com/scikit-image/data/-/raw/master/kidney-tissue-fluorescence.tif', ...}
shutil     = <module 'shutil' from '/usr/local/lib/python3.8/shutil.py'>
skimage_distribution_dir = '/opt/django-project/.tox/test-rgd-client/lib/python3.8/site-packages/skimage/data/..'
warn       = <built-in function warn>

../../.tox/test-rgd-client/lib/python3.8/site-packages/skimage/data/__init__.py:64: ImportError
```

</details>


Considering some high impact projects like scikit-image are now broken on import, it may need to be added back...

--- a/pooch/utils.py
+++ b/pooch/utils.py
@@ -50,7 +50,7 @@ def file_hash(*args, **kwargs):
     top-level namespace (`from pooch import file_hash`) instead, which is fully
     backwards compatible with pooch >= 0.1.
     """
-    warnings.warn(message, FutureWarning)
+    warnings.warn(message, DeprecationWarning)
     return new_file_hash(*args, **kwargs)
 
 

--- a/pooch/utils.py
+++ b/pooch/utils.py
@@ -14,6 +14,7 @@ import hashlib
 from pathlib import Path
 from urllib.parse import urlsplit
 from contextlib import contextmanager
+import warnings
 
 import appdirs
 from packaging.version import Version
@@ -21,6 +22,36 @@ from packaging.version import Version
 
 LOGGER = logging.Logger("pooch")
 LOGGER.addHandler(logging.StreamHandler())
+
+
+def file_hash(*args, **kwargs):
+    """
+    WARNING: Importing this function from pooch.utils is DEPRECATED.
+    Please import from the top-level namespace (`from pooch import file_hash`)
+    instead, which is fully backwards compatible with pooch >= 0.1.
+
+    Examples
+    --------
+
+    >>> fname = "test-file-for-hash.txt"
+    >>> with open(fname, "w") as f:
+    ...     __ = f.write("content of the file")
+    >>> print(file_hash(fname))
+    0fc74468e6a9a829f103d069aeb2bb4f8646bad58bf146bb0e3379b759ec4a00
+    >>> import os
+    >>> os.remove(fname)
+
+    """
+    # pylint: disable=import-outside-toplevel
+    from .hashes import file_hash as new_file_hash
+
+    message = """
+    Importing file_hash from pooch.utils is DEPRECATED. Please import from the
+    top-level namespace (`from pooch import file_hash`) instead, which is fully
+    backwards compatible with pooch >= 0.1.
+    """
+    warnings.warn(message, FutureWarning)
+    return new_file_hash(*args, **kwargs)
 
 
 def get_logger():

